packages/core/src/PluginManager.ts

```diff
@@ -6,11 +6,13 @@ export class PluginContext implements CanvacordPluginContext {
 
     register(name: string, options: { props: Record<string, Function> }): void {
         let methodNames = Object.keys(options.props);
-        Object.entries(options.props).forEach(
-            ([k, m]) =>
-                // @ts-ignore
-                (this.manager.core[k] = m)
-        );
+        Object.entries(options.props).forEach(([k, m]) => {
+            // @ts-ignore
+            if (this.manager.core[k] && !this.manager.core.options.overlapPlugins) throw new Error(`Cannot overlap "${k}" from ${name} plugin over ${this.manager.findPluginByMethod(k)} plugin.\nFix: Set [overlapPlugins] to true in options.`);
```

Comment:
this would also overwrite built-in props. Are you sure about that?

Reply:
```
let props = Object.keys(Object.getPrototypeOf(core));
props.push(...Object.keys(core));
this.addPlugin('@canvacord/core', props);
```

I was actually thinking of adding this in the constructor of `PluginManager`. Is there any other good idea?

Reply:
let's not allow overwriting existing props

Reply:
> 

Does this mean you cannot overlap props at all?